CommonTools/Utils/src/TMVAEvaluator.cc

```diff
@@ -0,0 +1,74 @@
+#include "CommonTools/Utils/interface/TMVAEvaluator.h"
+
+#include "CommonTools/Utils/interface/TMVAZipReader.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+
+TMVAEvaluator::TMVAEvaluator() :
+  mIsInitialized(false)
+{
+}
+
+
+TMVAEvaluator::~TMVAEvaluator()
+{
+}
+
+
+void TMVAEvaluator::initialize(const std::string & options, const std::string & method, const std::string & weightFile,
+                               const std::vector<std::string> & variables, const std::vector<std::string> & spectators)
+{
+  // initialize the TMVA reader
+  mReader.reset(new TMVA::Reader(options.c_str()));
+  mReader->SetVerbose(false);
+  mMethod = method;
+
+  // add input variables
+  for(std::vector<std::string>::const_iterator it = variables.begin(); it!=variables.end(); ++it)
+  {
+    mVariables.insert( std::pair<std::string,float>(*it,0.) );
+    mReader->AddVariable(it->c_str(), &mVariables.at(*it));
```

Comment:
When a map increases its size it does not affect the memory addresses for elements already in the map. Thus a map behaves differently than a vector when doing inserts.